lib/utils/flattenProperties.js

```diff
@@ -29,6 +29,15 @@ function flattenProperties(properties, to_ret) {
     if (properties.hasOwnProperty(name)) {
       // TODO: this is a bit fragile and arbitrary to stop when we get to a 'value' property.
       if (_.isPlainObject(properties[name]) && ('value' in properties[name])) {
+
+        /**
+         * If .value is omitted from the input
+         * ie value: "{color.core.purple}" instead of "{color.core.purple.value}"
+         */
+        if (properties[name].value.hasOwnProperty('value')) {
```

Comment:
I don't love putting this here, since this is flattenProperties function is supposed to be as generic as possible.